tfx/orchestration/experimental/kubernetes/orchestrator_container_entrypoint.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Main entrypoint for orchestrator container on Kubernetes."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import json
+import logging
+import sys
+import absl
+
+from google.protobuf import json_format
+from ml_metadata.proto import metadata_store_pb2
+from tfx.orchestration import pipeline
+from tfx.orchestration.experimental.kubernetes import kubernetes_dag_runner
+from tfx.utils import json_utils
+
+
+def main():
+  # Log to the container's stdout so it can be streamed by the client.
+  logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+  logging.getLogger().setLevel(logging.INFO)
+
+  parser = argparse.ArgumentParser()
+
+  # Pipeline is serialized via a json format.
+  # See kubernetes_remote_runner._serialize_pipeline for details.
+  parser.add_argument('--serialized_pipeline', type=str, required=True)
+  parser.add_argument('--tfx_image', type=str, required=True)
+  args = parser.parse_args()
+
+  tfx_pipeline = json.loads(args.serialized_pipeline)
+  components = [
+      json_utils.loads(component) for component in tfx_pipeline['components']
+    ]
+  metadata_connection_config = metadata_store_pb2.ConnectionConfig()
+  json_format.Parse(tfx_pipeline['metadata_connection_config'],
+                    metadata_connection_config)
```

Comment:
Could you refactor this deserialization logic (including the Pipeline object construction below) into its own method, `kubernetes_remote_runner._deserialize_pipeline()` and use that here?